test/unittest/StatisticsBackend/StatisticsBackendTests.cpp

```diff
@@ -16,13 +16,208 @@
 
 #include <StatisticsBackend.hpp>
 #include <types/types.hpp>
+#include "database/database.hpp"
 
 #include "gtest/gtest.h"
 
+#include <DatabaseUtils.hpp>
+#include <fastdds-statistics-backend/types/JSONTags.h>
+
 using namespace eprosima::statistics_backend;
+using namespace eprosima::statistics_backend::database;
+
+/**
+ * @brief Fixture for the statistics_backend class tests
+ *
+ * \c get_type Returns the entity kind of a given id.
+ *
+ */
+class statistics_backend_tests : public ::testing::Test
+{
+public:
+
+    typedef uint32_t TestId;
+
+    void SetUp()
+    {
+        entities = PopulateDatabase::populate_database(db);
+    }
+
+    DataBaseTest db;
+    std::map<TestId, std::shared_ptr<const Entity>> entities;
+};
+
+class StatisticsBackendTest : public StatisticsBackend
+{
+public:
+
+    static void set_database(
+            Database* db)
+    {
+        database_ = db;
+    }
+
+    static const char* get_entity_kind_str(
+            EntityKind kind)
+    {
+        return entity_kind_str[(int)kind];
+    }
+
+};
+
+void check_dds_entity(
+        std::shared_ptr<const DDSEntity> const& entity)
+{
+    Info info = StatisticsBackendTest::get_info(entity->id);
+    ASSERT_EQ(entity->guid, info[GUID_INFO_TAG]);
+    ASSERT_EQ(entity->qos, info[QOS_INFO_TAG]);
+}
+
+// Check the get_type StatisticsBackend method
+TEST_F(statistics_backend_tests, get_info)
+{
+    StatisticsBackendTest::set_database(&db);
+
+    // Check generic info
+    for (auto pair : entities)
+    {
+        std::shared_ptr<const Entity> entity = pair.second;
+        Info info = StatisticsBackendTest::get_info(entity->id);
+
+        ASSERT_EQ(entity->id, EntityId(info[ID_INFO_TAG]));
+        ASSERT_EQ(StatisticsBackendTest::get_entity_kind_str(entity->kind), info[KIND_INFO_TAG]);
+        ASSERT_EQ(entity->name, info[NAME_INFO_TAG]);
+    }
```

Comment:
Maybe we can check here the `EntityKind` and do the checking that are done below. Thus we avoid calling again `get_info` for the same entity twice. Improve the efficiency of the test.